### PR TITLE
[FIX] TT-1147 Uploads Disallowed for Data Freeze

### PR DESCRIPTION
**Pull request by huarngpa**
* Disallows file upload for `redacted` and `data_freeze`

### CODE REVIEW

File: sheepdog/utils/__init__.py [hunk 1]
```diff
@@ -498,6 +498,7 @@ def proxy_request(project_id, uuid, data, args, headers, method, action,
         return flask.Response(json.dumps({'message': message}), status=200)
 
     if action in ['upload', 'initiate_multipart']:
+        raise_for_data_freeze(indexd_doc)
```
Comment (anmaxvl) [inline sheepdog/utils/__init__.py]: for data freeze we should also block `delete` requests as well. I updated the ticket.

Comment (huarngpa) [reply sheepdog/utils/__init__.py]: Yeah we discussed and agreed this should be fine for non-admin

File: sheepdog/utils/__init__.py [hunk 2]
```diff
@@ -730,3 +732,13 @@ def f_retry(*args, **kwargs):
             return f(*args, **kwargs)
         return f_retry
     return deco_retry
+
+
+def raise_for_data_freeze(doc):
+    state = doc.metadata.get('state')
+    if state in ['redacted', 'data_freeze']:
```
Comment (kulgan) [inline sheepdog/utils/__init__.py]: Does the annotation endpoints currently handle the `data_freeze` too?

Comment (huarngpa) [reply sheepdog/utils/__init__.py]: No, this is preemptive

**Pull request merged**